### PR TITLE
Run an API call when verifying credentials

### PR DESCRIPTION
**Pull request by agrare**
Ensure that the VMM service is available in the region selected for the subscription provided.

Without this regions which we know are valid but not for certain subscriptions will return true for verify_credentials even when a subsequent refresh will fail.

For example I was able to successfully verify my credentials against "US Gov Virginia"

### CODE REVIEW

File: app/models/manageiq/providers/azure/manager_mixin.rb [hunk 1]
```diff
@@ -166,16 +173,14 @@ def raw_connect(client_id, client_key, azure_tenant_id, subscription, proxy_uri
 
       ::Azure::Armrest::Configuration.log = $azure_log
 
-      connection_rescue_block do
```
Comment (agrare) [inline app/models/manageiq/providers/azure/manager_mixin.rb:169]: The rescue block seems to make more sense for `self.verify_credentials` where we need to format the response to be shown on the UI, raw_connect shouldn't mess with exceptions IMO (only used elsewhere by the instance-level connect)

File: app/models/manageiq/providers/azure/manager_mixin.rb [hunk 2]
```diff
@@ -126,7 +128,7 @@ def params_for_create
     #   }
     # }
     def verify_credentials(args)
-      region           = args["region"]
+      region           = args["provider_region"]
```
Comment (Fryguy) [inline app/models/manageiq/providers/azure/manager_mixin.rb:131]: was this a bug?

Comment (agrare) [reply app/models/manageiq/providers/azure/manager_mixin.rb:131]: Yes, I have a separate PR to fix just this, the second commit in this branch builds on that fix

Comment (agrare) [reply app/models/manageiq/providers/azure/manager_mixin.rb:131]: https://github.com/ManageIQ/manageiq-providers-azure/pull/496

**Comment by Fryguy** [comment]
Backported to `morphy` in commit 57d1a4e7f88a1929c8ae8109f95b2871cb2636d6.

```text
commit 57d1a4e7f88a1929c8ae8109f95b2871cb2636d6
Author: Oleg Barenboim <chessbyte@gmail.com>
Date:   Wed Feb 23 15:19:47 2022 -0500

    Merge pull request #497 from agrare/run_simple_api_call_verify_credentials
    
    Run an API call when verifying credentials
    
    (cherry picked from commit 8e450435986b76d1457034b6c65a0cffdbd53264)
```

**Pull request merged**